api/src/main/java/io/opentelemetry/internal/Obfuscated.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.internal;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This interface allows the SDK to unobfuscate an obfuscated static global provider.
+ *
+ * <p>Static global providers are obfuscated when they are returned from the API to prevent users
+ * from casting them to their SDK specific implementation.
+ *
+ * <p>This is important for auto-instrumentation, because if users take the static global providers
+ * that are returned from the API, and cast them to their SDK specific implementations, then those
+ * casts will fail under auto-instrumentation, because auto-instrumentation takes over the static
+ * global providers returned by the API and points them to it's embedded SDK.
+ *
+ * @since 0.4.0
+ */
+@ThreadSafe
+public interface Obfuscated {
+
+  /**
+   * Returns the unobfuscated provider.
+   *
+   * @return the unobfuscated provider.
+   * @since 0.4.0
+   */
+  Object unobfuscate();
```

Comment:
Rather than `Object`, can we have this interface be parameterized with a type?

Reply:
I like that idea. Done.

Reply:
@trask before doing this change, can we clarify why having this interface at all is necessary?

Reply:
@bogdandrutu just wanted to make sure that you saw my response https://github.com/open-telemetry/opentelemetry-java/pull/1152#discussion_r416841711 (I should have replied here also)